types/main.d.ts

```diff
@@ -0,0 +1,624 @@
+declare module "pdfjs" {
+    export class Image {
+        /**
+         * Creates a new image that can be added to one or multiple documents.
+         * @param src a buffer of the image source (supported formats: jpeg, pdf)
+         */
+        constructor(src: Buffer | ArrayBuffer);
+    }
+
+    export class Font {
+        /**
+         * Creates a new AFM font pr OTF font object that can be used with PDF documents.
+         * TFont objects can be used multiple times.
+         * @param arg the font data, as either Buffer or ArrayBuffer of an OTF font or a JSON object descriping an AFM font
+         */
+        constructor(arg: Buffer | ArrayBuffer | FontDescription);
```

Comment:
What I like most about this PR is, that it makes me rethink certain parts of the API. The user shouldn't ever provide a `FontDescription` theirselves, but this is not _enforced_ by the API. So instead of `new pdf.Font(require('pdfjs/font/Helvetica.json'))` it should have really been `require('pdfjs/font/Helvetica.js')` (which is an instance of `AFMFont`. No action required here, just something that needs to improve later on.